cmd/commands/git-source.go

```diff
@@ -0,0 +1,204 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/codefresh-io/cli-v2/pkg/log"
+	"github.com/codefresh-io/cli-v2/pkg/runtime"
+	"github.com/codefresh-io/cli-v2/pkg/store"
+	"github.com/codefresh-io/cli-v2/pkg/util"
+
+	"github.com/argoproj-labs/argocd-autopilot/pkg/application"
+	"github.com/argoproj-labs/argocd-autopilot/pkg/fs"
+	"github.com/argoproj-labs/argocd-autopilot/pkg/git"
+	apstore "github.com/argoproj-labs/argocd-autopilot/pkg/store"
+	wf "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow"
+	wfv1alpha1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	"github.com/go-git/go-billy/v5/memfs"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type (
+	GitSourceCreateOptions struct {
+		insCloneOpts *git.CloneOptions
+		gsCloneOpts  *git.CloneOptions
+		gsName       string
+		runtimeName  string
+		fullGsPath   string
+	}
+)
+
+func NewGitSourceCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "git-source",
+		Short: "Manage git-sources of Codefresh runtimes",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.HelpFunc()(cmd, args)
+			exit(1)
+		},
+	}
+
+	cmd.AddCommand(NewGitSourceCreateCommand())
+
+	return cmd
+}
+
+func NewGitSourceCreateCommand() *cobra.Command {
+	var (
+		insCloneOpts *git.CloneOptions
+		gsCloneOpts  *git.CloneOptions
+	)
+
+	cmd := &cobra.Command{
+		Use:   "create runtime_name git-source_name git-src-repo_full_path",
+		Short: "add a new git-source to an existing runtime",
+		Example: util.Doc(`
+			<BIN> git-source create runtime_name git-source-name https://github.com/owner/repo-name/my-workflow
+		`),
+		PreRun: func(cmd *cobra.Command, args []string) {
+			ctx := cmd.Context()
+
+			if len(args) < 1 {
+				log.G(ctx).Fatal("must enter runtime name")
+			}
+
+			if len(args) < 2 {
+				log.G(ctx).Fatal("must enter git-source name")
+			}
+
+			if gsCloneOpts.Repo == "" {
+				log.G(ctx).Fatal("must enter a valid value to --git-src-repo. Example: https://github.com/owner/repo-name/path/to/workflow")
+			}
+
+			if gsCloneOpts.Auth.Password == "" {
+				gsCloneOpts.Auth.Password = insCloneOpts.Auth.Password
+			}
+
+			insCloneOpts.Parse()
+			gsCloneOpts.Parse()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			err := RunCreateGitSource(ctx, &GitSourceCreateOptions{
```

Comment:
just return the RunCreateGitSource function call
the errors inside print good error messages to the user